watertap/unit_models/thickener.py

```diff
@@ -221,11 +221,15 @@ def TSS_in(blk, t):
 
         @self.Expression(self.flowsheet().time, doc="Thickening factor")
         def f_thick(blk, t):
-            return blk.p_thick * (10 / (blk.TSS_in[t]))
+            eps = 1e-30 * pyunits.kg / pyunits.m**3
+            return blk.p_thick * (10 / (blk.TSS_in[t] + eps))
 
         @self.Expression(self.flowsheet().time, doc="Remove factor")
         def f_q_du(blk, t):
-            return blk.TSS_rem / (pyunits.kg / pyunits.m**3) / 100 / blk.f_thick[t]
+            eps = 1e-30 * pyunits.m**3 / pyunits.kg
+            return (
+                blk.TSS_rem / (pyunits.kg / pyunits.m**3) / 100 / (blk.f_thick[t] + eps)
```

Comment:
Would bounding `f_thick` resolve this issue? As it stands you are adding an effective bound of 1e-30 (which I will note is effectively meaningless unless you have some massive scaling factors applied).

Reply:
`f_thick` is an Expression, so I can't apply a bound to it directly. And adding a constraint to bound it seems like it could be more trouble than it's worth

Reply:
Well, that touches on the heart of the issue. To avoid evaluation errors you need to reformulate issues at the root cause. So here your options are to reformulate the constraint that is using this Expression such that `f_thick` is moved to the other side of the equation if possible, or you really do need some sort of intermediate variable to keep things bounded (or decide it is an acceptable issue which is sometimes unavoidable).

Reply:
Alright, in this situation I can just rearrange it such that f_thick is moved to the other side, but in cases where it may not be possible to do this, is it really better to deem it an "acceptable issue" rather than just adding an `eps` term? Ig what I'm getting at is when should you add an `eps` term?

Reply:
My suggestion is that you should generally not add the `eps` as it could potentially cause results to change. If your only solution is the add an `eps` then you have basically already concluded the issue is unavoidable and are just making an arbitrary change to satisfy the diagnostics check. In that case, you would just be better leaving a note that these are known, unavoidable issues.